conans/test/functional/generators/premake_test.py

```diff
@@ -0,0 +1,77 @@
+import os
+import textwrap
+import unittest
+
+from nose.plugins.attrib import attr
+
+from conans import load
+from conans.client.tools import chdir, which
+from conans.test.utils.tools import TestClient
+
+
+@attr("slow")
+@attr("premake")
+@unittest.skipIf(which("premake5") is None, "Needs premake5")
+class PremakeGeneratorTest(unittest.TestCase):
+
+    def setUp(self):
+        self.client = TestClient()
+        self.client.run("new test/1.0")
+        self.client.run("create . danimtb/testing")
+        conanfile = textwrap.dedent("""
+        [requires]
+        test/1.0@danimtb/testing
+        
+        [generators]
+        premake
+        """)
+        premake = textwrap.dedent("""
+        include("conanbuildinfo.lua")
+
+        workspace("example")
+            conan_basic_setup()
+
+            project("example")
+            kind "ConsoleApp"
+            language "C++"
+            targetdir = "bin/%{cfg.buildcfg}"
+
+            files{
+                "src/**",
+            }
+
+            filter "configurations:Debug"
+                defines { "DEBUG" }
+                symbols "On"
+
+            filter "configurations:Release"
+                defines { "NDEBUG" }
+                optimize "On"
+        """)
+        hello_cpp = textwrap.dedent("""
+        #include "hello.h"
+        
+        int main()
+        {
+            hello();
+        }
+        """)
+        self.client.save({"conanfile.txt": conanfile,
+                          "premake5.lua": premake,
+                          "src/hello.cpp": hello_cpp}, clean_first=True)
+
+    def test_generate_basic_setup_release(self):
+        self.client.run("install .")
```

Comment:
If I run this test in a x86 machine, will it fail? I mean, if Conan autodetects `x86`... the checks below talks about `x86_64`:

```python
self.assertIn("Release|x64", sln_content)
self.assertNotIn("Debug|x64", sln_content)
```

Reply:
It is highly unlikely that test are run on a native 32 bit machine and Release is the auto-detect default so I think there is no risk here

Reply:
We don't want to have a false negative in that case, this test has nothing to do with autodetect functionality. It is better to have a `-s arch=x86_64` in the `install` call.